internal/cloud/realm/templates.go

```diff
@@ -65,3 +70,24 @@ func (c *client) ClientTemplate(groupID, appID, templateID string) (*zip.Reader,
 
 	return zipPkg, nil
 }
+
+func (c *client) CompatibleTemplates(groupID, appID string) ([]Template, bool, error) {
```

Comment:
hmm i think both places are probably valid in some sense. I was in a hurry when I reviewed last time, so I misinterpreted your comment. This is on me.

for this endpoint, we return 400 if the current app was not created using a template app. The client template handler does this too.

Because 400's are "bad requests", it makes sense that we should stop the command and tell the user that their command is incorrect. We should halt wherever we encounter 400 errors.

Reply:
then it would seem to me we want to error the whole command in that case, and instruct the user what went wrong, rather than proceed with a command to behave in the way they didn't specify

much like in the UI if we see any status code returned outside of `[200, 300)` we throw an error, I would expect the same here

Reply:
I think that us returning a 400 from `baas` makes sense because we shouldn't be trying to get a ClientTemplate for an app that wasn't made by a template. But in the CLI, we don't know if the app was made with or without a template unless we ask `baas`, which is why we're using the 400 response code to check.  

I think if we had a call where we could check, like `isTemplateApp` or something that returns 200 with either true or false, then we can avoid making the ClientTemplate call check for it.

Reply:
so just to clarify, I am less questioning the use of `400` for this response, but more so why are we swallowing that error here in the CLI?

if its a true error (and it sounds like it is from both descriptions here), then we should treat it as such.  provide feedback to the user about what went wrong: `"the app you requested was not created via a template..."` (you have a better idea on the messaging for something like that than i).  we can even go as far to provide suggested fixes: "remove the `--template` flag", "please use a templated app", etc

Reply:
Yeah, I see. I think during some iterations ago, I might've used the `nil` return value for something, but looking at how the `resolveClientTemplates` evolved now, it might not be useful to have the `nil` value. 

I think it wouldn't be so bad in the current workflow to have the error being thrown, but @andrewzlchen let me know what you think.

Reply:
yeah throwing the error sounds good to me!